src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenReadOnlySpanConstructionTest.cs

```diff
@@ -2685,5 +2652,779 @@ public static int M()
+using System;
+using System.Runtime.CompilerServices;
+
+public class C
+{
+    [DateTimeConstant(-1)] static DateTime dateTime = default;
+
+    public static System.ReadOnlySpan<DateTime> M()
+        => new DateTime[] { dateTime };
+}
+""";
+            var compilation = CreateCompilationWithMscorlibAndSpan(src);
+            var verifier = CompileAndVerify(compilation, verify: Verification.Skipped);
+            var expectedIL = """
+{
+  // Code size       24 (0x18)
+  .maxstack  4
+  IL_0000:  ldc.i4.1
+  IL_0001:  newarr     "System.DateTime"
+  IL_0006:  dup
+  IL_0007:  ldc.i4.0
+  IL_0008:  ldsfld     "System.DateTime C.dateTime"
+  IL_000d:  stelem     "System.DateTime"
+  IL_0012:  call       "System.ReadOnlySpan<System.DateTime> System.ReadOnlySpan<System.DateTime>.op_Implicit(System.DateTime[])"
+  IL_0017:  ret
+}
+""";
+            verifier.VerifyIL("C.M", expectedIL);
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/69472")]
+        public void ReadOnlySpanFromArrayOfConstants_DateTime_WithConst()
+        {
+            var src = """
+using System;
+using System.Runtime.CompilerServices;
+
+public class C
+{
+    [DateTimeConstant(-1)] const DateTime dateTime = default;
+}
+""";
+            var compilation = CreateCompilationWithMscorlibAndSpan(src);
+            compilation.VerifyDiagnostics(
+                // (6,28): error CS0283: The type 'DateTime' cannot be declared const
+                //     [DateTimeConstant(-1)] const DateTime dateTime = default;
+                Diagnostic(ErrorCode.ERR_BadConstType, "const").WithArguments("System.DateTime").WithLocation(6, 28),
+                // (6,54): error CS0133: The expression being assigned to 'C.dateTime' must be constant
+                //     [DateTimeConstant(-1)] const DateTime dateTime = default;
+                Diagnostic(ErrorCode.ERR_NotConstantExpression, "default").WithArguments("C.dateTime").WithLocation(6, 54)
+                );
+        }
+
+        [Theory, WorkItem("https://github.com/dotnet/roslyn/issues/69472")]
+        [InlineData("sbyte")]
+        [InlineData("byte")]
+        [InlineData("short")]
+        [InlineData("ushort")]
+        [InlineData("int")]
+        [InlineData("uint")]
+        [InlineData("long")]
+        [InlineData("ulong")]
+        [InlineData("float")]
+        [InlineData("double")]
+        public void ReadOnlySpanFromArrayOfConstants_IntegerTypesUsingBlob(string type)
+        {
+            var src = $$"""
+var values = C.M();
+System.Console.Write($"{values.Length} {values[0]}");
+
+public class C
+{
+    public static System.ReadOnlySpan<{{type}}> M()
+        => new {{type}}[] { 42 };
+}
+""";
+            var compilation = CreateCompilationWithMscorlibAndSpan(src);
+            CompileAndVerify(compilation, expectedOutput: "1 42", verify: Verification.Skipped).VerifyDiagnostics();
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/69472")]
```

Comment:
>WorkItem("https://github.com/dotnet/roslyn/issues/69472") [](http://example.com/codeflow?start=15&length=57)

Did the behavior of this test or the next test change with this PR?

Reply:
No, these two tests behave the same as before. This was to ensure there was no impact on existing scenarios. We probably have these two covered already in some fashion, but it was easier this way